applications/CoSimulationApplication/python_scripts/solver_wrappers/rigid_body/rigid_body_solver_wrapper.py

```diff
@@ -0,0 +1,67 @@
+# Importing the Kratos Library
+import KratosMultiphysics as KM
+import KratosMultiphysics.CoSimulationApplication as KMC
+
+# Importing the base class
+from KratosMultiphysics.CoSimulationApplication.base_classes.co_simulation_solver_wrapper import CoSimulationSolverWrapper
+
+# Other imports
+from .rigid_body_solver import RigidBodySolver
+
+def Create(settings, model, solver_name):
+    return RigidBodySolverWrapper(settings, model, solver_name)
+
+class RigidBodySolverWrapper(CoSimulationSolverWrapper):
```

Comment:
it seems like this class does not specify that the RBS solver only runs in one rank (I am guessing it is not MPI parallel is it)

By default each solver uses all available MPI cores:
https://github.com/KratosMultiphysics/Kratos/blob/20fa5829e9a94db266972fc94339b46e3c690746/applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_solver_wrapper.py#L150-L152

Hence in this case you need to specify that it only runs on one rank. This is done by providing a `DataCommunicator` that provides only one rank, see what I did in the SDOF solver.
https://github.com/KratosMultiphysics/Kratos/blob/20fa5829e9a94db266972fc94339b46e3c690746/applications/CoSimulationApplication/python_scripts/solver_wrappers/sdof/sdof_solver_wrapper.py#L64-L66

Otherwise each rank will have its own instance of this class which I am guessing can interfere with e.g. output or other stuff

@RiedlAndreas this is what I tried to explain in the Mail. I am guessing/assuming you use the new RBS solver?

Reply:
Done, thanks!

There was no problem in the output before, since the output functions were done only in rank 0:

https://github.com/KratosMultiphysics/Kratos/blob/b07afc0a8890d5819f7c9fd5b3354b70f2c4f1ca/applications/CoSimulationApplication/python_scripts/solver_wrappers/rigid_body/rigid_body_solver.py#L187-L189

Should we erase this check now that the solver is going to run entirely in rank 0 anyway or is it better to leave the double check?